src/dialect/onnx/onnx_ops.cpp

```diff
@@ -868,9 +871,31 @@ void ONNXConvNoBiasOp::inferShapes() {
     // K - 1
     //
     // where K is a kernel spatial dimension.
-    // Pad as if stride is 1.
-    for (int i = 0; i < nDims; ++i)
-      outSpatialDims[i] += kernelDims[i] - 1;
+    // Take into consideration the stride.
+    auto strides = ONNXConvNoBiasOp::stridesAttr();
+    for (int i = 0; i < nDims; ++i) {
+      // If strides are given use them otherwise stride is 1.
+      int64_t stride = 1;
+      if (strides)
+        stride = ArrayAttrIntVal(strides, i);
+
+      // Compute necessary padding.
+      int64_t totalPadding = stride * (dataShape[i + 2] - 1) +
+          kernelDims[i] - dataShape[i + 2];
+
+      // Adjust current output value with the value of the padding.
+      // When dividing by the number of slides later on, the output
+      // dimension should be equal to the input dimension.
+      outSpatialDims[i] += totalPadding;
+
+      // Record the upper and lower axis padding.
+      actualPads[i] = actualPads[i + nDims] = totalPadding / 2;
+      if (autoPad == "SAME_LOWER") {
+        actualPads[i]++;
```

Comment:
Not sure what happens if totalPadding is a even number and rounding up is not needed?